### DISCUSSION TITLE
[Feature] Option to delete all assets from a stack

### DISCUSSION THREAD

**Discussion by townsste** [immich-app/immich#5656] [Feature Request]
### The bug

When you delete a stacked photo (JPG and RAW) they go to trash.  When you go to trash and empty trash or delete the stacked photos, it only seems to delete the JPG and the RAW file shows back up on the app and also does persist in the location folder.  

You must then go back and delete the RAW image as done above, in the app, to fully remove the two photos.  As a result, you need to double delete instead of being able to delete both stacked photos at one time.

This also happens when you disable trash on the server.  Only the cover or the JPG from the stacked photo is deleted.  When you refresh photos the RAW file reappears and you need to delete that now to.

**Comment by shenlong-tanwen** [comment]
@townsste This is currently this way by design. Since we allow manually stacking assets, the user might not actually want to remove all the assets but the only one currently viewed. Probably a separate option to remove the entire stack could be a different feature. But currently, sharing, deleting, favorites, archive, etc are all asset specific actions and are applied to the entire stack

**Reply by StevenMassaro** [reply]
The main issue here is that if you are on the first item in a stack, and press delete, only that item from the stack is moved to the trash, and the other items in the stack are no longer visible on the main photos UI. They have disappeared to invisible place, and are only visible again once you empty the trash.

The workaround is to delete the other items in the stack first, then delete the top item. This is pretty inconvenient though. It would be fine if pressing delete while on the top item in the stack kept the other stack items in the photos page, so at least you could go delete them.

**Comment by townsste** [comment]
Oh... Gotcha.  I understand.  I do like the idea of adding this as a feature then for faster deletion of images.  Because if a stack is good and you want to delete the images then you have to go back and delete the data twice. Once for jpg then again for raw.

I used immich-go to automatically stack my jpg and raw files together, after being uploaded to immich server, and then once stacked I thought I would be able to delete the entire stack in one go. 

For my case, if I want to delete the jpg then I also want to delete the raw.  By being able to delete the stack then it speeds up the process of removing unwanted jpg and corresponding raw.

Thank you for the information.

**Reply by shenlong-tanwen** [reply]
I've converted the issue to a FR discussion now
